network/nxos/nxos_vxlan_vtep_vni.py

```diff
@@ -0,0 +1,566 @@
+                existing_value = existing_commands.get(key)
+                if key == 'peer-ip':
+                    for peer in existing_value:
+                        commands.append('no {0} {1}'.format(key, peer))
+                else:
+                    commands.append('no {0} {1}'.format(key, existing_value))
+            else:
+                if key.replace(' ', '_').replace('-', '_') in BOOL_PARAMS:
+                    commands.append('no {0}'.format(key.lower()))
+        else:
+            command = '{0} {1}'.format(key, value.lower())
+            commands.append(command)
+
+    if commands:
+        vni_command = 'member vni {0}'.format(module.params['vni'])
+        ingress_replication_command = 'ingress-replication protocol static'
+        interface_command = 'interface {0}'.format(module.params['interface'])
+
+        if ingress_replication_command in commands:
+            static_level_cmds = [cmd for cmd in commands if 'peer' in cmd]
+            parents = [interface_command, vni_command, ingress_replication_command]
+            candidate.add(static_level_cmds, parents=parents)
+            commands = [cmd for cmd in commands if 'peer' not in cmd]
+
+        if vni_command in commands:
+            parents = [interface_command]
+            commands.remove(vni_command)
+            if module.params['assoc_vrf'] is None:
+                parents.append(vni_command)
+            candidate.add(commands, parents=parents)
+
+
+def state_absent(module, existing, proposed, candidate):
+    if existing['assoc_vrf']:
+        commands = ['no member vni {0} associate-vrf'.format(
+                                                module.params['vni'])]
+    else:
+        commands = ['no member vni {0}'.format(module.params['vni'])]
+    parents = ['interface {0}'.format(module.params['interface'])]
+    candidate.add(commands, parents=parents)
+
+
+def main():
+    argument_spec = dict(
+            interface=dict(required=True, type='str'),
+            vni=dict(required=True, type='str'),
+            assoc_vrf=dict(required=False, type='bool'),
+            multicast_group=dict(required=False, type='str'),
+            peer_list=dict(required=False, type='list'),
+            suppress_arp=dict(required=False, type='bool'),
+            ingress_replication=dict(required=False, type='str',
+                                     choices=['bgp', 'static', 'default']),
+            m_facts=dict(required=False, default=False, type='bool'),
+            state=dict(choices=['present', 'absent'], default='present',
+                       required=False),
+            include_defaults=dict(default=True),
+            config=dict(),
+            save=dict(type='bool', default=False)
+    )
+    module = get_network_module(argument_spec=argument_spec,
+                                supports_check_mode=True)
+
+    if module.params['assoc_vrf']:
+        mutually_exclusive_params = ['multicast_group',
```

Comment:
Please look at examples of `mutually_exclusive`, this is built in validation,  e.g cloud/openstack/_keystone_user.py